### DISCUSSION TITLE
Which pulsar version is recommended for use in production environments?

### DISCUSSION THREAD

**Discussion by a327919006** [apache/pulsar#20806] [Q&A] [answered]
I am planning to use Pulsar in a production environment and would like to know which version is more stable and suitable for use in the production environment.Thanks.

3.0.x
2.11.x
2.10.x
2.9.x
2.8.x

**Comment by tisonkun** [comment, marked as answer]
Hi @a327919006!

You can refer to https://pulsar.apache.org/contribute/release-policy/ to see the version policy of Pulsar.

Briefly, 3.0.x is the first LTS version and will be supported in the next 30 months. 2.10.x, different from noted in the page linked above, is the earliest maintained versions at the moment whose broker side is compatible with JDK 11 (2.11 and 3.0 support JDK 17+).

However, all version of Pulsar client support JDK 8+ and you can use a Docker image to deploy Pulsar cluster where JDK is bundled.
